.github/workflows/check-go-validation.yml

```diff
@@ -0,0 +1,26 @@
+name: "[Check] go validation"
+on:
+  push:
+    branches: [ "master" ]
```

Comment:
Can we enable running this action for pushes to all branches?

It makes development easier, since then contributors can see that CI is failing before they make a PR, and so they don't need to litter the PR with a bunch of force-pushes.

```suggestion
```

Reply:
I think if somebody is contributing then it is still from another repository and the action would be from that fork. As for pushing to this repository, I think it is good enough to have it for pr and push to master only.

Reply:
:+1: It does make contributing from a fork a bit easier if actions run on all branches.

But then again, from a fork, it's not too hard to make a commit enabling it, or just force-pushing to a PR if there is a failure.

Reply:
@vdovhanych it costs us nothing, so let's do it? I can do it.